File: 백준/Silver/2839. 설탕 배달/설탕 배달.rs
fn main() {
    let mut input = String::new();
    std::io::stdin().read_line(&mut input).unwrap();
    let mut num = input.trim().parse::<i32>().unwrap();
    let mut answer: i32 = 0;

    while num >= 0 {
        if num % 5 == 0 {
            answer += num / 5;
            println!("{}", answer);
            return;
        }

        num -= 3;
        answer += 1;
    }
    println!("{}", -1);
}

File: 백준/Silver/2839. 설탕 배달/설탕 배달.py
import sys

input = sys.stdin.readline


def answer(N):
    answer = 0
    while N >= 0:
        if N % 5 == 0:
            answer += N // 5
            return answer
        N -= 3
        answer += 1
    return -1


if __name__ == "__main__":
    N = int(input())
    print(answer(N))
